src/chain.ts

```diff
@@ -14,37 +14,148 @@
  *  OTHER TORTIOUS ACTION, ARISING OUT OF OR IN CONNECTION WITH THE USE OR
  *  PERFORMANCE OF THIS SOFTWARE.
  */
-import { AE_AMOUNT_FORMATS, formatAmount } from './utils/amount-formatter'
-import verifyTransaction from './tx/validator'
-import { pause } from './utils/other'
-import { isNameValid, produceNameId, decode } from './tx/builder/helpers'
-import { DRY_RUN_ACCOUNT, AensName } from './tx/builder/schema'
+import { AE_AMOUNT_FORMATS, formatAmount } from './utils/amount-formatter';
+import verifyTransaction, { ValidatorResult } from './tx/validator';
+import { pause } from './utils/other';
+import { isNameValid, produceNameId } from './tx/builder/helpers';
+import { DRY_RUN_ACCOUNT } from './tx/builder/schema';
+import { AensName } from './tx/builder/constants';
 import {
-  AensPointerContextError, DryRunError, InvalidAensNameError, InvalidTxError,
-  RequestTimedOutError, TxTimedOutError, TxNotInChainError, InternalError
-} from './utils/errors'
-import Node, { TransformNodeType } from './Node'
+  AensPointerContextError, DryRunError, InvalidAensNameError, TransactionError,
+  TxTimedOutError, TxNotInChainError, InternalError,
+} from './utils/errors';
+import Node, { TransformNodeType } from './Node';
 import {
   Account as AccountNode, ByteCode, ContractObject, DryRunResult, DryRunResults,
-  Generation, KeyBlock, MicroBlockHeader, NameEntry, SignedTx
-} from './apis/node'
-import { EncodedData } from './utils/encoder'
-import AccountBase from './account/Base'
+  Generation, KeyBlock, MicroBlockHeader, NameEntry, SignedTx,
+} from './apis/node';
+import { decode, Encoded, Encoding } from './utils/encoder';
+import AccountBase from './account/Base';
 
-export function _getPollInterval (
+/**
+ * @category chain
+ */
+export function _getPollInterval(
   type: 'block' | 'microblock',
   { _expectedMineRate = 180000, _microBlockCycle = 3000, _maxPollInterval = 5000 }:
-  { _expectedMineRate?: number, _microBlockCycle?: number, _maxPollInterval?: number }
+  { _expectedMineRate?: number; _microBlockCycle?: number; _maxPollInterval?: number },
 ): number {
   const base = {
     block: _expectedMineRate,
-    microblock: _microBlockCycle
-  }[type]
-  return Math.min(base / 3, _maxPollInterval)
+    microblock: _microBlockCycle,
+  }[type];
+  return Math.min(base / 3, _maxPollInterval);
+}
+
+/**
+ * @category exception
+ */
+export class InvalidTxError extends TransactionError {
```

Comment:
Wouldn't it be more concise to keep this under `utils/errors`?

Reply:
Sometimes errors (like InvalidTxError) depend on functionality or types defined in other modules, and keeping them in `utils/errors` leads to cycle dependencies that complicates testing and not working in some environments.

In general I'm trying to migrate from grouping functionality by programming entities (like `tx/builder/constants` and 
 `utils/errors`) to group it by problem it solves, for example https://github.com/aeternity/aepp-sdk-js/blob/4d203c6eecb525dfb8fa8ac84add2f607e77b1ee/src/tx/builder/field-types/fee.ts contains all functionality to calculate fee with minimal exports. Before it was splitted into constants (that actually not needed to other code) and a helper function that were all exported.